server/util/lockingbuffer/lockingbuffer.go

```diff
@@ -8,14 +8,19 @@ import (
 
 // LockingBuffer is a thread-safe buffer.
 type LockingBuffer struct {
-	buffer bytes.Buffer
-	mu     sync.RWMutex
+	buffer  bytes.Buffer
```

Comment:
nit: if we added `LimitedBuffer` like I suggested in the other comment, then we could make this an interface instead of a `bytes.Buffer` (e.g. `type WrappedBuffer interface { io.Reader, io.Writer, Len() int }`), which would be satisfied by both `*bytes.Buffer` and `*commandutil.LimitedBuffer`). Then, we wouldn't have to add complexity to the implementation by copying the limiting logic. (I'd be fine either way though.)